sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -487,7 +487,7 @@ public void write(final ChannelHandlerContext context, final Object message, fin
             final RntbdRequestRecord record = (RntbdRequestRecord) message;
             this.timestamps.channelWriteAttempted();
 
-            context.write(this.addPendingRequestRecord(context, record), promise).addListener(completed -> {
+            context.writeAndFlush(this.addPendingRequestRecord(context, record), promise).addListener(completed -> {
```

Comment:
This was something I have asked before and was told is not needed.
Can you please expand on why is it needed and how is it related to ConnectionStateListener.

Reply:
Agreed. Thanks for catching that. It is not required and the change was likely the result of an experiment that I was running. I flush further down the pipeline.

Reply:
change is on my working branch